driver-sync/src/main/com/mongodb/client/FindIterable.java

```diff
@@ -331,4 +331,50 @@ public interface FindIterable<TResult> extends MongoIterable<TResult> {
      * @mongodb.server.release 3.2
      */
     <E> E explain(Class<E> explainResultClass, ExplainVerbosity verbosity);
+
+    /**
+     * Explain the execution plan for this operation.
+     *
+     * @param timeoutMS the timeout in milliseconds for the explain operation
+     * @return the execution plan
+     * @mongodb.driver.manual reference/command/explain/
+     * @mongodb.server.release 3.2
+     */
+    Document explain(long timeoutMS);
```

Comment:
I'm not clear why all these overloads are needed.  For the rest of CSOT, applications specify the timeout via `MongoCluster/MongoDatabase/MongoCollection#withTimeout`. Why doesn't that suffice for explain?